src/dump.c

```diff
@@ -2014,9 +2014,13 @@ DLLEXPORT jl_value_t *jl_uncompress_ast(jl_lambda_info_t *li, jl_value_t *data)
 
 DLLEXPORT int jl_save_incremental(const char *fname, jl_array_t *worklist)
 {
+    // make temporary file name foo.ji.XXXXXX
+    // (would be better to use mkstemp once this is in libuv -- see libuv/libuv#322)
+    char *tmpfname = mktemp(strcat(strcpy((char *) malloc(strlen(fname)+8), fname), ".XXXXXX"));
```

Comment:
if we use `.lock`, could we also `flock` it to prevent multiple copies of julia trying to build the same library at the same time? it may be better to handle that at the julia level in `Base.cachecompile`, however.